sunpy/net/dataretriever/sources/kanzelhohe.py

```diff
@@ -0,0 +1,128 @@
+    ...                       a.Wavelength(6563*u.AA))   #doctest: +REMOTE_DATA
+    >>> print(results)  #doctest: +REMOTE_DATA
+    Results from 1 Provider:
+    <BLANKLINE>
+    1 Results from the KanzelhoheClient:
+         Start Time           End Time      Source   Instrument      Wavelength
+    ------------------- ------------------- ------ -------------- ---------------
+    2015-12-28 00:00:00 2015-12-28 09:03:00    KSO Kanzelhohe HA2 6563.0 Angstrom
+    <BLANKLINE>
+    <BLANKLINE>
+    >>> response = Fido.fetch(results)  #doctest: +SKIP
+    """
+
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Return list of URLS corresponding to value of input timerange.
+
+        Parameters
+        ----------
+        timerange: `~sunpy.time.TimeRange`
+            time range for which data is to be downloaded.
+
+        Returns
+        -------
+        urls : `list`
+            list of URLs corresponding to the requested time range
+        """
+        wave = kwargs['wavelength']
+        wave_float = (wave.min if isinstance(wave, a.Wavelength) else wave.min()).value
+        table = {6563: {'datatype': ['halpha2k/recent', 'halph_fr'],
+                        'start_time': datetime.datetime(2000, 7, 20, 7, 45, 46)},
+                 32768: {'datatype': ['caiia', 'caiik_fi'],
+                         'start_time': datetime.datetime(2000, 7, 20, 7, 45, 46)},
+                 5460: {'datatype': ['phokada', 'bband_fi'],
+                        'start_time': datetime.datetime(2000, 7, 20, 7, 45, 46)}}
+        # Checking if value is close enough to a wavelength value.
+        # Converting from one unit to other introduces precision errors.
+        wavelengths = table.keys()
+        wave_list = list(filter(lambda x: np.isclose(x, wave_float), wavelengths))
+        if(len(wave_list) == 0):
+            raise ValueError("no value is enough closer to allowed wavelengths: "
+                             + ",".join(map(str, table.keys())) + " angstroms")
+        wave = wave_list[0]
+        start_date = table[wave]['start_time']
+        if timerange.start < start_date:
+            msg = 'Earliest date for which Kanzelhohe data is available is {:%Y-%m-%d}'
+            raise ValueError(msg.format(start_date))
+        prefix = "http://cesar.kso.ac.at/{datatype}/%Y/"
+        suffix = ""
+        if wave != 6563:
+            suffix = "%Y%m%d/processed/"
+        url_pattern = prefix + suffix + "kanz_{datatype1}_%Y%m%d_%H%M%S.fts.gz"
+        dtypes = table[wave]['datatype']
+        scraper = Scraper(url_pattern, datatype=dtypes[0], datatype1=dtypes[1])
+        if not timerange:
+            return []
+        result = scraper.filelist(timerange)
+        return result
+
+    def _makeimap(self):
+        """
+        Helper Function:used to hold information about source.
+        """
+        self.map_['source'] = 'KSO'  # TODO: check with kanzelhohe
+        self.map_['instrument'] = 'Kanzelhohe HA2'
```

Comment:
```suggestion
        self.map_['instrument'] = 'Kanzelhöhe HA2'
```

Reply:
Should I remove todo in Line 103?

Reply:
I wouldn't. In any case, once we merge it we may create an issue about it too.